hooks/tk-multi-publish2/photoshopcc.basic/upload_version.py

```diff
@@ -0,0 +1,348 @@
+# Copyright (c) 2017 Shotgun Software Inc.
+# 
+# CONFIDENTIAL AND PROPRIETARY
+# 
+# This work is provided "AS IS" and subject to the Shotgun Pipeline Toolkit 
+# Source Code License included in this distribution package. See LICENSE.
+# By accessing, using, copying or modifying this work you indicate your 
+# agreement to the Shotgun Pipeline Toolkit Source Code License. All rights 
+# not expressly granted therein are reserved by Shotgun Software Inc.
+
+import os
+import pprint
+import tempfile
+import uuid
+import sgtk
+
+HookBaseClass = sgtk.get_hook_baseclass()
+
+
+class PhotoshopUploadVersionPlugin(HookBaseClass):
+    """
+    Plugin for sending photoshop documents to shotgun for review.
+    """
+
+    @property
+    def icon(self):
+        """
+        Path to an png icon on disk
+        """
+
+        # look for icon one level up from this hook's folder in "icons" folder
+        return os.path.join(
+            self.disk_location,
+            os.pardir,
+            "icons",
+            "review.png"
+        )
+
+    @property
+    def name(self):
+        """
+        One line display name describing the plugin
+        """
+        return "Send document to Shotgun review"
+
+    @property
+    def description(self):
+        """
+        Verbose, multi-line description of what the plugin does. This can
+        contain simple html for formatting.
+        """
+        return """
+        This plugin exports and uploads a copy of the Photoshop document to
+        Shotgun for review.
+        """
+
+    @property
+    def settings(self):
+        """
+        Dictionary defining the settings that this plugin expects to recieve
+        through the settings parameter in the accept, validate, publish and
+        finalize methods.
+
+        A dictionary on the following form::
+
+            {
+                "Settings Name": {
+                    "type": "settings_type",
+                    "default": "default_value",
+                    "description": "One line description of the setting"
+            }
+
+        The type string should be one of the data types that toolkit accepts as
```

Comment:
Reading this, I still don't know what should be given as the type. Maybe a couple of examples?